parsers/midi_parser.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+"""
+Translates any folder of MIDI files to a set of JAMS files.
+
+To parse the MIDI files, you just need to provide the path to the folder
+containing the files.
+
+Example:
+./midi_parser.py ~/midi_files -o ~/midi_jams/
+
+NOTE: pitch bend information is ignored, so notes with pitch bends will be
+converted to notes with a fixed pitch value.
+
+"""
+
+__author__ = "J. Salamon"
+__copyright__ = "Copyright 2016, Music and Audio Research Lab (MARL)"
+__license__ = "GPL"
+__version__ = "1.0"
+__email__ = "justin.salamon@nyu.edu"
+
+import argparse
+import logging
+import os
+import time
+import midi
+import pretty_midi
+import pandas as pd
+
+import jams
+
+
+def fill_file_metadata(jam, midi_file, duration):
+    """Fills the global metada into the JAMS jam."""
+    jam.file_metadata.title = os.path.basename(midi_file)[:-4]
+    jam.file_metadata.duration = duration
+
+
+def fill_annotation_metadata(annot):
+    """Fills the annotation metadata."""
+    annot.annotation_metadata.corpus = ""
+    annot.annotation_metadata.version = ""
+    annot.annotation_metadata.annotation_tools = ""
+    annot.annotation_metadata.annotation_rules = ""
+    annot.annotation_metadata.validation = ""
+    annot.annotation_metadata.data_source = ""
+    annot.annotation_metadata.curator = jams.Curator(name="",
+                                                     email="")
+    annot.annotation_metadata.annotator = {}
+
+
+def create_jams(midi_file, out_file):
+    """
+    Creates a JAMS file from a MIDI file
+    """
+
+    # Create jam
+    jam = jams.JAMS()
+
+    # Create annotation
+    midi_ann = jams.Annotation('pitch_midi')
+
+    # Load midi file
+    pm = pretty_midi.PrettyMIDI(midi_file)
+
+    notes = []
+
+    for inst in pm.instruments:
+        for note in inst.notes:
+            midi_ann.append(time=note.start, duration=(note.end-note.start),
```

Comment:
midi files are generally messy, it's non-trivial to group tracks into instruments (as @craffel can attest), and the mapping between tracks<--> instruments is not 1-1. So this is the simplest solution, which I'd keep as the default behavior. That said, I don't see why we couldn't provide an optional parameter that would allow you to ask for each instrument to go in a separate annotation (where instruments are defined using the grouping used in `pretty_midi`), and if you really think it'd be useful we could consider creating an annotation per-track too (messier).